File: Java/pieceofcake2.java
/*
 * Problem: Piece of Cake!
 * 
 */

import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;

public class pieceofcake2 {

	public static void main(String[] args) {
		final int thickness = 4;
		
		Scanner sc = new Scanner(System.in);
	
		int length = sc.nextInt();
		int horizDist = sc.nextInt();
		int vertDist = sc.nextInt();

		int volPieceA = horizDist * vertDist * thickness;
		int volPieceB = horizDist * (length - vertDist) * thickness;
		int volPieceC = (length - horizDist) * vertDist * thickness;
		int volPieceD = (length - horizDist) * (length - vertDist) * thickness;
		
		Integer[] volArr = {volPieceA, volPieceB, volPieceC, volPieceD};
		int maxVolume = Collections.max(Arrays.asList(volArr));
		
		System.out.println(maxVolume);
	} //end main

}

File: Python/pieceofcake2.py
'''
 *
 *	Problem: "Piece of Cake!"
 *
'''

import sys
import math

THICKNESS = 4

line = input().split()

length = float( line[0] )
horizDist = float( line[1] )
vertDist = float( line[2] )

volPieceA = horizDist * vertDist * THICKNESS;
volPieceB = horizDist * (length - vertDist) * THICKNESS;
volPieceC = (length - horizDist) * vertDist * THICKNESS;
volPieceD = (length - horizDist) * (length - vertDist) * THICKNESS;
		
maxVolume = max(volPieceA, volPieceB, volPieceC, volPieceD)
print(round(maxVolume))